frontend/src/api/network/servingRuntimes.ts

```diff
@@ -11,80 +11,66 @@ import { ServingRuntimeKind } from '../../k8sTypes';
 import { CreatingServingRuntimeObject } from 'pages/modelServing/screens/types';
 import { getModelServingRuntimeName } from 'pages/modelServing/utils';
 import { getModelServingProjects } from './projects';
+import { getConfigMap } from './configMaps';
+import { DEFAULT_MODEL_SERVING_TEAMPLATE } from 'pages/modelServing/screens/const';
+import YAML from 'yaml';
+
+const fetchServingRuntime = (
+  data: CreatingServingRuntimeObject,
+  configNamespace: string,
+  namespace: string,
+): Promise<ServingRuntimeKind> => {
+  return getConfigMap(configNamespace, 'servingruntimes-config')
+    .then((configmap) => {
+      const servingRuntime = YAML.parse(configmap.data?.['default-config'] || '');
+      return assembleServingRuntime(data, servingRuntime, namespace);
+    })
+    .catch(() => {
+      const servingRuntime = DEFAULT_MODEL_SERVING_TEAMPLATE;
+      return assembleServingRuntime(data, servingRuntime, namespace);
+    });
+};
 
 const assembleServingRuntime = (
   data: CreatingServingRuntimeObject,
+  servingRuntime: ServingRuntimeKind,
   namespace: string,
 ): ServingRuntimeKind => {
   const { numReplicas, modelSize, externalRoute, tokenAuth } = data;
   const name = getModelServingRuntimeName(namespace);
+  const updatedServingRuntime = { ...servingRuntime };
 
-  return {
-    apiVersion: 'serving.kserve.io/v1alpha1',
-    kind: 'ServingRuntime',
-    metadata: {
+  updatedServingRuntime.metadata = {
+    name,
+    namespace,
+    labels: {
+      ...servingRuntime.metadata.labels,
       name,
-      namespace,
-      labels: {
-        name,
-        'opendatahub.io/dashboard': 'true',
-      },
-      annotations: {
-        ...(externalRoute && { 'enable-route': 'true' }),
-        ...(tokenAuth && { 'enable-auth': 'true' }),
-      },
+      'opendatahub.io/dashboard': 'true',
     },
-    spec: {
-      supportedModelFormats: [
-        {
-          name: 'openvino_ir',
-          version: 'opset1',
-          autoSelect: true,
-        },
-        {
-          name: 'onnx',
-          version: '1',
-          autoSelect: true,
+    annotations: {
+      ...servingRuntime.metadata.annotations,
+      ...(externalRoute && { 'enable-route': 'true' }),
+      ...(tokenAuth && { 'enable-auth': 'true' }),
+    },
+  };
+  updatedServingRuntime.spec.replicas = numReplicas;
+  updatedServingRuntime.spec.containers = servingRuntime.spec.containers.map((container) => {
```

Comment:
There is no safety for `spec` not being there... In my previous comment, you'll spread `null` and then get a NPE on `.spec` when doing `.replicas`

Curious if you think just setting `servingRuntime` to `DEFAULT_MODEL_SERVING_TEAMPLATE` if it's null or undefined will just solve this?

Reply:
mmmm this is a great discussion, I've addressed the issue with servingRuntime being null, but as now it will be a user configurable spec inside a configmap the user can misconfigured the object in many ways. With the current block, if there's a misconfiguration in the key spec elements (the ones we are checking in this function) it will fail and fallback to `DEFAULT_MODEL_SERVING_TEAMPLATE `.
But this is a tricky question, having a ConfigMap is not the best solution IMO, everything could go south since there's not "training wheels" for this feature, we might wanna check this in the future.

Reply:
We'll need to log a bug to handle this more and surface it to the user more -- please log one -- I want to make it a blocker for 1.22 / aka our next ODH Dashboard release

Reply:
Done here: https://github.com/opendatahub-io/odh-dashboard/issues/893